routing/probability_estimator.go

```diff
@@ -123,6 +123,22 @@ func (p *probabilityEstimator) getPairProbability(
 	)
 }
 
+// getLocalPairProbability estimates the probability of successfully traversing
+// our own local channels to toNode.
+func (p *probabilityEstimator) getLocalPairProbability(
+	now time.Time, results NodeResults, toNode route.Vertex) float64 {
+
+	// For local channels that have never been tried before, we assume them
+	// to be successful. We have accurate balance and online status
+	// information on our own channels, so when we select them in a route it
+	// is close to certain that those channels will work.
+	nodeProbability := p.prevSuccessProbability
```

Comment:
The call to `calculateProbability` below looks at the history for this channel. But the important difference with a remote channel is that we consider an untried channel and a previously successful local channel as equal.

The actual probability that we use here hardly matters, because it is a cost factor that is added to every route. To stay close to the probability concept, 1 is maybe not the best choice. Because also local channels - even though we have more info - aren't guaranteed to be successful.

Maybe there should be a diff between untried and successful local chans. But the difference will probably be so small, that we can consider them equal.

Reply:
thanks for the explanation, i agree the difference is probably small. sgtm